### PR TITLE
Fix running WebAssembly Browser App from VS

### LINKED ISSUE #75356: Unable to run WebAssembly Browser App from VS or CLI
### Description

I'm unable to run a new WebAssembly Browser App from VS or the CLI. 

### Reproduction Steps

```
dotnet workload install wasm-experimental
dotnet new wasmbrowser
dotnet run
```

Or create the project from VS and try to run in VS.

### Expected behavior

App runs successfully

### Actual behavior

The application to execute does not exist: 'C:\Program'

### Regression?

_No response_

### Known Workarounds

_No response_

### Configuration

7.0.100-rc.1.22429.3
VS 17.4 P1

### Other information

_No response_

### PR DESCRIPTION
**Pull request by thaystg**
As `$(OutputPath)` is not absolute path, we need to add `$(MSBuildProjectDirectory)`

Tested:

- Create a new WebAssembly Browser App on VS and run, change WasmAppDir and run again.
- Create a new WebAssembly Console App on VS and run, change WasmAppDir and run again.
- Create a new wasmbrowser app using command line and run;
- Create a new wasmconsole app using command line and run;
- Create a new wasmbrowser app using command line and dotnet build and dotnet run --no-build;
- Create a new wasmbrowser app using command line and dotnet build and dotnet run --debug;

Also, added tests for command line runs. But a few of the tests are disabled because the tests would be using 7.0 targets, but 7.0 doesn't have the fix yet.

Fix #75356

### CODE REVIEW

File: src/mono/wasm/build/WasmApp.targets [hunk 1]
```diff
@@ -117,10 +117,7 @@
...
                      the RID. So, we don't have the correct final OutputPath here. But we need it for `dotnet run` to work,
                      as it just reads the RunCommand after evaluation. -->
-    <_AppBundleDirForRunCommand Condition="Exists('$(WasmAppDir)/$(AssemblyName).runtimeconfig.json')">$(WasmAppDir)</_AppBundleDirForRunCommand>
-    <_AppBundleDirForRunCommand Condition="'$(_AppBundleDirForRunCommand)' == '' and Exists('$(OutputPath)/AppBundle')">$([MSBuild]::NormalizeDirectory($(OutputPath), 'AppBundle'))</_AppBundleDirForRunCommand>
-    <_AppBundleDirForRunCommand Condition="'$(_AppBundleDirForRunCommand)' == '' and Exists('$(OutputPath)/browser-wasm/AppBundle')">$([MSBuild]::NormalizeDirectory($(OutputPath), 'browser-wasm', 'AppBundle'))</_AppBundleDirForRunCommand>
-    <_AppBundleDirForRunCommand Condition="'$(_AppBundleDirForRunCommand)' == ''">OutputPath=$(OutputPath), OutDir=$(OutDir)</_AppBundleDirForRunCommand>
+    <_AppBundleDirForRunCommand Condition="'$(_AppBundleDirForRunCommand)' == ''">$([MSBuild]::NormalizeDirectory($(MSBuildProjectDirectory), $(OutputPath), 'browser-wasm', 'AppBundle'))</_AppBundleDirForRunCommand>
```
Comment (radical) [inline src/mono/wasm/build/WasmApp.targets:120]: We should check that `OutputPath` is not rooted, and only then assign prepend this.

**Comment by radical** [comment]
The corresponding 7.0 PR has been merged. Once the packages from that are available, the disabled test here can be enabled again.

**Comment by lewing** [comment]
failure is https://github.com/dotnet/runtime/issues/74302

**Review by lewing — APPROVED** [review]

**Pull request merged**